src/Jwt/LcobucciFactory.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\Jwt;
+
+use Lcobucci\JWT\Configuration;
+use Lcobucci\JWT\Signer;
+use Lcobucci\JWT\Signer\Key;
+use Symfony\Component\Mercure\Exception\InvalidArgumentException;
+
+if (!class_exists(Key\InMemory::class)) {
+    throw new \LogicException('You cannot use "Symfony\Component\Mercure\Token\LcobucciFactory" as the "lcobucci/jwt" package is not installed. Try running "composer require lcobucci/jwt".');
+}
+
+final class LcobucciFactory implements TokenFactoryInterface
+{
+    /**
+     * @var array<string, class-string<Signer>>
+     */
+    public const SIGN_ALGORITHMS = [
+        'hmac.sha256' => Signer\Hmac\Sha256::class,
+        'hmac.sha384' => Signer\Hmac\Sha384::class,
+        'hmac.sha512' => Signer\Hmac\Sha512::class,
+        'ecdsa.sha256' => Signer\Ecdsa\Sha256::class,
+        'ecdsa.sha384' => Signer\Ecdsa\Sha384::class,
+        'ecdsa.sha512' => Signer\Ecdsa\Sha512::class,
+        'rsa.sha256' => Signer\Rsa\Sha256::class,
+        'rsa.sha384' => Signer\Rsa\Sha384::class,
+        'rsa.sha512' => Signer\Rsa\Sha512::class,
+    ];
+
+    private $configurations;
+    private $claims;
+
+    /**
+     * @param array<string, mixed> $claims
+     */
+    public function __construct(string $secret, string $algorithm = 'hmac.sha256', array $claims = [])
+    {
+        if (!\array_key_exists($algorithm, self::SIGN_ALGORITHMS)) {
+            throw InvalidArgumentException::forInvalidAlgorithm($algorithm, \array_keys(self::SIGN_ALGORITHMS));
+        }
+
+        $signerClass = self::SIGN_ALGORITHMS[$algorithm];
+        $this->configurations = Configuration::forSymmetricSigner(
+            new $signerClass(),
+            Key\InMemory::plainText($secret)
+        );
+
+        $this->claims = $claims;
+    }
+
+    public function create(array $publish, array $subscribe): string
+    {
+        $builder = $this->configurations
+            ->builder()
+            ->withClaim('mercure', [
+                'publish' => $publish,
+                'subscribe' => $subscribe,
+            ]);
+
+        foreach ($this->claims as $name => $value) {
+            $builder = $builder->withClaim($name, $value);
```

Comment:
Note that calling `withClaim()` with a standard claim will throw (e.g. exp, iss, iat.. full list available at https://github.com/lcobucci/jwt/blob/4.0.x/src/Token/RegisteredClaims.php). The `Builder`'s claim-specific methods need be used for those.
Btw, does this factory add any standard claim by default?

Reply:
It should at lease support `exp`. By the way `exp` cannot be passed in the constructor because the date will most likely change for each generation.

Reply:
Hm, what do you suggest here?

Reply:
fixed - additional claims are now allowed via `create` method arguments.